### PR TITLE
[Core] Fix worker crash when getting actor name from runtime context

### LINKED ISSUE #45174: [Worker] Worker crash when getting an actor name
### What happened + What you expected to happen

Expect to return an empty name or raise an exception, but the worker crash.

```
[2024-05-07 11:37:17,261 C 3465 7571412] actor_manager.cc:47:  Check failed: it != actor_handles_.end() Cannot find an actor handle of id, NIL_ID. This method should be called only when you ensure actor handles exists.
*** StackTrace Information ***
0   _raylet.so                          0x0000000107dcc2bc _ZN3raylsERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEERKNS_10StackTraceE + 84 ray::operator<<()
1   _raylet.so                          0x0000000107df2e88 _ZN3ray13SpdLogMessage5FlushEv + 220 ray::SpdLogMessage::Flush()
2   _raylet.so                          0x0000000107df2cf0 _ZN3ray13SpdLogMessageD2Ev + 20 ray::SpdLogMessage::~SpdLogMessage()
3   _raylet.so                          0x0000000107dcf928 _ZN3ray6RayLogD2Ev + 52 ray::RayLog::~RayLog()
4   _raylet.so                          0x00000001073ff050 _ZNK3ray4core12ActorManager14GetActorHandleERKNS_7ActorIDE + 704 ray::core::ActorManager::GetActorHandle()
5   _raylet.so                          0x000000010742dae0 _ZNK3ray4core10CoreWorker12GetActorNameEv + 56 ray::core::CoreWorker::GetActorName()
6   _raylet.so                          0x00000001073017bc _ZL52__pyx_pw_3ray_7_raylet_10CoreWorker_23get_actor_nameP7_objectS0_ + 32 __pyx_pw_3ray_7_raylet_10CoreWorker_23get_actor_name()
7   python3.8                           0x0000000104b2afe4 method_vectorcall_NOARGS + 312 method_vectorcall_NOARGS
8   python3.8                           0x0000000104c28824 _PyEval_EvalFrameDefault + 30612 _PyEval_EvalFrameDefault
9   python3.8                           0x0000000104b1b7ac _PyFunction_Vectorcall + 416 _PyFunction_Vectorcall
10  python3.8                           0x0000000104b29cd8 property_descr_get + 108 property_descr_get
11  python3.8                           0x0000000104b7e32c _PyObject_GetMethod + 648 _PyObject_GetMethod
12  python3.8                           0x0000000104c28fcc _PyEval_EvalFrameDefault + 32572 _PyEval_EvalFrameDefault
13  python3.8                           0x0000000104c1f1ac _PyEval_EvalCodeWithName + 736 _PyEval_EvalCodeWithName
14  python3.8                           0x0000000104c915e0 run_mod + 172 run_mod
15  python3.8                           0x0000000104c8ff84 pyrun_file + 156 pyrun_file
16  python3.8                           0x0000000104c8fa14 pyrun_simple_file + 352 pyrun_simple_file
17  python3.8                           0x0000000104c8f874 PyRun_SimpleFileExFlags + 64 PyRun_SimpleFileExFlags
18  python3.8                           0x0000000104cb228c pymain_run_file + 264 pymain_run_file
19  python3.8                           0x0000000104cb18e8 pymain_run_python + 360 pymain_run_python
20  python3.8                           0x0000000104cb1728 Py_RunMain + 40 Py_RunMain
21  python3.8                           0x0000000104aee1a0 main + 56 main
22  dyld                                0x00000001820da0e0 start + 2360 start

```

### Versions / Dependencies

Ray 2.10.0

### Reproduction script

```python
import ray

if __name__ == "__main__":
    ray.init("local")
    context = ray.get_runtime_context()
    print(context.worker.actor_name())
```

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by 982945902**
## Why are these changes needed?

Currently calling get_runtime_context().get_actor_name() from driver will crash. Instead of crashing, this PR returns None  in this case.

## Related issue number
Closes #45174

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by kevin85421 — COMMENTED** [review]
How about updating `runtime_context.py` instead of `core_worker.cc` to keep the behavior consistent with other functions such as `get_task_id` and `get_actor_id`?

To elaborate, 

```python
    def get_actor_name(self) -> Optional[str]:
        """Get the current actor name of this worker.

        This shouldn't be used in a driver process.
        The name is in string format.

        Returns:
            The current actor name of this worker.
            If a current worker is an actor, and
            if actor name doesn't exist, it returns an empty string.
            If a current worker is not an actor, it returns None.
        """
        # only worker mode has actor_id
        if self.worker.mode != ray._private.worker.WORKER_MODE:
            logger.warning(
                "This method is only available when the process is a "
                f"worker. Current mode: {self.worker.mode}"
            )
            return None <---- Add this line
        actor_id = self.worker.actor_id
        return self.worker.actor_name if not actor_id.is_nil() else None
```

You can refer to [get_task_id](https://github.com/ray-project/ray/blob/master/python/ray/runtime_context.py#L241-L260) as an example.

**Comment by 982945902** [comment]
@kevin85421  that good !

**Review by kevin85421 — COMMENTED** [review]
Would you mind adding a test in `test_runtime_context.py`? Btw, you can add `-s` in your `git commit` command to avoid the DCO failure in CI.

**Comment by 982945902** [comment]
@kevin85421  ok

File: doc/source/_templates/navbar-anyscale.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<a id="try-anyscale-href" href="https://forms.gle/DFin3AzKEDB19uXZ6">
+<a id="try-anyscale-href" href="https://www.anyscale.com">
```
Comment (aslonnie) [inline doc/source/_templates/navbar-anyscale.html:1]: this line change is unrelated? could you rebase properly?

Comment (982945902) [reply doc/source/_templates/navbar-anyscale.html:1]: ok

Comment (982945902) [reply doc/source/_templates/navbar-anyscale.html:1]: https://github.com/ray-project/ray/pull/45016/files  from this?

**Comment by kevin85421** [comment]
Would you mind rebasing with the master branch?

**Comment by 982945902** [comment]
> Would you mind rebasing with the master branch?

@kevin85421  not mind, how need i do

**Comment by kevin85421** [comment]
Take a look at your commit history. It seems to be that you don't sync with the master branch correctly. You may not need to rebase with the master branch, but you need to fix your commit history.

**Comment by kevin85421** [comment]
Hi @982945902, would you mind fixing the CI failure?

**Comment by 982945902** [comment]
@kevin85421  貌似有些测试用例发生了超时，造成用例未通过，我重新触发C试下

**Comment by kevin85421** [comment]
> It seems that some of the test cases timed out, causing the use cases to fail, how should I re-triggered the C test

I re-triggered CI pipeline multiple times for this PR. It always fails. Could you try to reproduce in your local env?

**Comment by 982945902** [comment]
@kevin85421  These test cases have bugs, and someone merged and fixed them two days ago

**Comment by 982945902** [comment]
@kevin85421  These test cases have bugs, and someone merged and fixed them two days ago

**Comment by kevin85421** [comment]
Would you mind rebasing with the master branch? Thanks!

**Comment by 982945902** [comment]
> Would you mind rebasing with the master branch? Thanks!

@kevin85421  not mind, how need i do

**Comment by kevin85421** [comment]
> @kevin85421 not mind, how need i do

Sync your branch with the upstream master branch.

**Comment by 982945902** [comment]
@kevin85421  done

**Review by kevin85421 — APPROVED** [review]
